pycue/tests/cloud/gce_api_test.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+
+#  Copyright Contributors to the OpenCue Project
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import mock
+import unittest
+import os
+
+import opencue
+import opencue.cloud.gce_api
+
+TEST_CLOUD_GROUP_NAME = "test-group-main"
+TEST_INSTANCE_TEMPLATE_NAME = "test-instance-template"
+
+
+class GoogleCloudManagerTest(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    @mock.patch.object(opencue.cloud.gce_api.GoogleCloudManager, "connect", autospec=True)
+    def test_get_all_groups(self, connect_mock):
+        # __getitem__.side_effect is used to make the return value of mock
+        # list call subscriptable since the actual code uses it like a dictionary
+        test_response = {
+            "items": [
+                {
+                    "name": TEST_CLOUD_GROUP_NAME,
+                    "id": 1234
+                }
+            ]
+        }
+        connect_mock.return_value = None
```

Comment:
Do you need this `return_value`? If you just want to prevent connect from being called, mocking it should be sufficient unless the calling code requires a specific response.

If you just need to prevent something from being called and don't need to set up a return value, you can set it up like:

```
@mock.patch.object(opencue.cloud.gce_api.GoogleCloudManager, "connect", autospec=True, new=mock.MagicMock(return_value=False))
```

(with or without `return_value`, depending on what's needed).

This will prevent the mock from being passed in as an input to the test method, which can keep the code a bit cleaner.

Reply:
Yes, that makes sense. After moving `connect` out of the constructor I wouldn't even need to patch it like you mentioned. I forgot to move it out of the tests. Thanks for catching that!